### PR TITLE
Get routersteps mq 4

### PR DESCRIPTION
**Pull request by mitchell-mccracken**
Closes MQ-4

Solution:

* Added content to `routeStep/controller.ts` file including `allRouteSteps` function.
* Added `routeStep` path to `app.ts`

### CODE REVIEW

File: src/routeStep/controller.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+module.exports = router;
+
+//routes
+router.get('/', allRouteSteps);
+
+
+
+//functions
+function allRouteSteps(_req: any, res: any) {
```
Comment (jarrilla) [inline src/routeStep/controller.ts:15]: types

File: src/routeStep/controller.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import express = require('express');
+import { ExpressHandler } from '../utils';
+const router = express.Router();
+import { RouteStepModel } from './model';
+
+
+module.exports = router;
```
Comment (jarrilla) [inline src/routeStep/controller.ts:7]: don't use `module.exports` in module files.
Instead you should do
```
const router = Router();
export default router;
```

**Review by jarrilla — CHANGES_REQUESTED** [review]
Lint and add to tech doc, please.

File: tests/unit/routeSteps.spec.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+const url = '/routeSteps';
+
+// set up dbUrl
+let dbUrl = process.env.MONGO_DB_URI;
+dbUrl = dbUrl!.substring(0, dbUrl!.lastIndexOf('--TEST') )
+
+const client = new MongoClient( dbUrl, { useUnifiedTopology: true } );
+
+async function clientConnect() {
```
Comment (jarrilla) [inline tests/unit/routeSteps.spec.ts:15]: doesnt' need to be a function
you can just write
```
await CLIENT.connect().catch(console.error);
```

in line 29

**Review by jarrilla — CHANGES_REQUESTED** [review]

File: tests/unit/routeSteps.spec.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+import { describe, it } from "mocha";
+import { ChaiRequest } from "../config";
+import { MongoClient } from 'mongodb';
+require('../config'); // recommended way of loading root hooks
+
+const url = '/routeSteps';
+
+// set up dbUrl
+let dbUrl = process.env.MONGO_DB_URI;
+dbUrl = dbUrl!.substring(0, dbUrl!.lastIndexOf('--TEST') )
```
[the remark this answers is not in the document]
Comment (mitchell-mccracken) [reply tests/unit/routeSteps.spec.ts]: removed

File: tests/unit/routeSteps.spec.ts [hunk 5]
```diff
@@ -0,0 +1,44 @@
+import { expect } from "chai";
+import { describe, it } from "mocha";
+import { ChaiRequest } from "../config";
+import { MongoClient } from 'mongodb';
+require('../config'); // recommended way of loading root hooks
+
+const url = '/routeSteps';
```
[the remark this answers is not in the document]
Comment (mitchell-mccracken) [reply tests/unit/routeSteps.spec.ts]: updated

File: tests/unit/routeSteps.spec.ts [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+// set up dbUrl
+let dbUrl = process.env.MONGO_DB_URI;
```
[the remark this answers is not in the document]
Comment (mitchell-mccracken) [reply tests/unit/routeSteps.spec.ts]: updated

File: tests/unit/routeSteps.spec.ts (hunk 3 above)
Comment (mitchell-mccracken) [reply tests/unit/routeSteps.spec.ts]: addressed as suggested

**Review by jarrilla — APPROVED** [review]

**Pull request merged**